Bindings refactoring & Automap bindings

Ch0wW opened the pull request:
Refactored the bindings, so that all new binding commands won't need to be copy/pasted.
Also added independant automap bindings, and added a menu control dedicated to it.

Ch0wW committed:
Rewrote binding class

Ch0wW committed:
Add "tilde" to the list of bindings.

Also put back uparrow and backarrow for vanilla preservation.

Ch0wW committed:
Made Automapbinds (Fixes bug 1092)

Also :
- Made a check about the "togglemap" command, in case of an accidental automap key rebound.
- Checks if IDDT works.
- Created a cvar called am_followplayer, and added Action commands.

BUG Post : https://odamex.net/bugs/show_bug.cgi?id=1092

Ch0wW committed:
Fix empty binds being randomly written in the CFG.

Also renamed automapbind to ambind for now, I'm unsure of how to call them.

Ch0wW committed:
Added Automap bindings in the menu.

Ch0wW committed:
Update m_options.cpp

AlexMax commented:
Nice work.  My only concern is that if you upgrade from an older version of Odamex and have an existing config, none of the automap keys will be bound by default.

Ch0wW committed:
Modified "Unbindall" to allow automap binds to populate.

Ch0wW commented:
The last commit should rectify the issue you mentionned.

Ch0wW committed:
Merge branch 'development' of https://github.com/odamex/odamex into bindings_rewrite

Ch0wW committed:
Merge remote-tracking branch 'upstream/development-protobreak' into bindings_rewrite

Ch0wW committed:
Fix building.

AlexMax commented on client/src/c_bind.cpp:
```diff
@@ -570,4 +549,94 @@ std::string C_GetKeyStringsFromCommand(const char *cmd, bool bTwoEntries)
+		Bindings.UnbindKey(argv[1]);
+	}
+}
+END_COMMAND(unbind)
```
Could you add a short little help if argc < 2 instead of printing nothing?

AlexMax commented on client/src/c_bind.cpp:
```diff
@@ -570,4 +549,94 @@ std::string C_GetKeyStringsFromCommand(const char *cmd, bool bTwoEntries)
+		DoubleBindings.UnbindKey(argv[1]);
+	}
+}
+END_COMMAND(undoublebind)
```
Could you add a short little help if argc < 2 instead of printing nothing?

AlexMax commented on client/src/c_bind.cpp:
```diff
@@ -570,4 +549,94 @@ std::string C_GetKeyStringsFromCommand(const char *cmd, bool bTwoEntries)
+		AutomapBindings.UnbindKey(argv[1]);
+	}
+}
+END_COMMAND(amunbind)
```
Could you add a short little help if argc < 2 instead of printing nothing?

leximayfield reviewed (changes requested):
Mostly looks good, just have a few small things to take care of and I'll merge ASAP.

Ch0wW committed:
Tweak some elements:
- Added usage description for unbind, undoublebind, unambind
- Removed GetBinding
- Made GetBind as a const std::string&
- Renamed amunbind to unambind

Ch0wW committed:
Added "all" parameter to unbind commands.

Ch0wW committed:
Correct the "zoom in" key for the automap.

AlexMax commented:
LGTM
